lang/java/avro/src/main/java/org/apache/avro/file/ZstandardCodec.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.avro.file;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import com.github.luben.zstd.Zstd;
+import com.github.luben.zstd.ZstdInputStream;
+import com.github.luben.zstd.ZstdOutputStream;
+
+/**
+ * Implements zstd (https://github.com/facebook/zstd) compression
+ * using the jni wrappers from https://github.com/luben/zstd-jni.
+ * <p/>
+ * Supports compression levels from 1 to 22, with level 1 similar
+ * to gzip/deflate in compression ratio, but significantly faster.
+ * High levels of compression get close to xz compression ratios
+ * with decompression speeds 10x faster than xz.
+ */
+public class ZstandardCodec extends OutputInputStreamCodec {
+
+  static class Option extends CodecFactory {
+    private final int compressionLevel;
+
+    Option(int compressionLevel) {
+      this.compressionLevel = compressionLevel;
+    }
+
+    @Override
+    protected Codec createInstance() {
+      return new ZstandardCodec(compressionLevel);
+    }
+  }
+
+  private final int compressionLevel;
+
+  public ZstandardCodec(int compressionLevel) {
+    this.compressionLevel = Math.max(Math.min(compressionLevel, 22), 1);
```

Comment:
note zstandard now has negative compression levels, for use cases that require even less CPU use: https://github.com/facebook/zstd/releases/tag/v1.3.4